v2/tools/generator/internal/codegen/pipeline/add_operator_spec.go

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/config"
+)
+
+const AddOperatorSpecStageID = "addOperatorSpec"
+
+func AddOperatorSpec(configuration *config.Configuration, idFactory astmodel.IdentifierFactory) Stage {
+	return MakeStage(
+		AddOperatorSpecStageID,
+		"Adds the property 'OperatorSpec' to all Spec types that require it",
+		func(ctx context.Context, state *State) (*State, error) {
+			defs := state.Types()
+			result := make(astmodel.Types)
+
+			propInjector := astmodel.NewPropertyInjector()
+			resources := astmodel.FindResourceTypes(defs)
+
+			for _, resource := range resources {
```

Comment:
Suggest pulling much of the body of this loop out into a method or two, to make it declaratively clear what the loop does.

Possibly `NeedsOperatorSpec(rsrc ResourceType) bool` and `CreateOperatorSpec(rsrc ResourceType) PropertyDefinition`
or maybe just `TryCreateOperatorSpec(rsrc ResourceType) (*PropertyDefinition, err)`

Reply:
Pulled most of the loop body into `createOperatorSpecIfNeeded(configuration, idFactory, resolved) (astmodel.Types, error)`